rescan.go

```diff
@@ -562,22 +562,13 @@ func rescan(chain ChainSource, options ...RescanOption) error {
 		blockFilter, err := chain.GetCFilter(
 			newStamp.Hash, wire.GCSFilterRegular, queryOptions,
 		)
-
-		switch {
-		// If the block index doesn't know about this block, then it's
-		// likely we're mid re-org so we'll accept this as we account
-		// for it below.
-		case err == headerfs.ErrHashNotFound:
-
-		case err != nil:
-			return fmt.Errorf("unable to get filter for hash=%v: %v",
-				curStamp.Hash, err)
-		}
-
-		// If the filter is nil, then this either means that we don't
-		// have any peers to fetch this filter from, or the peer(s) that
-		// we're trying to fetch from are in the progress of a re-org.
-		if blockFilter == nil {
+		if err != nil {
+			// If the query failed, then this either means that we
+			// don't have any peers to fetch this filter from, or
+			// the peer(s) that we're trying to fetch from are in
+			// the progress of a re-org.
+			log.Errorf("unable to get filter for hash=%v, "+
```

Comment:
Perhaps just log the error if it fails with something other than `ErrFilterNotFound`? Can imagine it getting spammy in some cases.

Reply:
You mean in cases where we fail to fetch the filter because we have no peers/timeout etc? Shouldn't be a very expected error though?

Reply:
Just thinking it's more likely in a mobile scenario where users are switching networks, losing reception, etc.

Reply:
Mhm, i think it is worth logging though

Reply:
Nit: logged error should start capitalized.